crates/bevy_animation/src/graph.rs

```diff
@@ -172,6 +183,96 @@ pub enum AnimationGraphLoadError {
     SpannedRon(#[from] SpannedError),
 }
 
+/// Acceleration structures for animation graphs that allows Bevy to evaluate
+/// them quickly.
+///
+/// These are kept up to date as [`AnimationGraph`] instances are added,
+/// modified, and removed.
+#[derive(Default, Reflect, Resource)]
+pub struct ThreadedAnimationGraphs(
+    pub(crate) HashMap<AssetId<AnimationGraph>, ThreadedAnimationGraph>,
+);
+
+/// An acceleration structure for an animation graph that allows Bevy to
+/// evaluate it quickly.
+///
+/// This is kept up to date as the associated [`AnimationGraph`] instance is
+/// added, modified, or removed.
+#[derive(Default, Reflect)]
+pub(crate) struct ThreadedAnimationGraph {
```

Comment:
Would it be useful to expose this publicly to users, so that if they store animation graphs in somewhere other than `Assets<AnimationGraph>`, they can still use this (with some more involved manual setup)?